jbmc/src/java_bytecode/remove_instanceof.cpp

```diff
@@ -256,7 +256,10 @@ bool remove_instanceoft::lower_instanceof(
   }
 
   return lower_instanceof(
-           function_identifier, target->code_nonconst(), goto_program, target) |
+           function_identifier,
+           target->code_nonconst(),
+           goto_program,
+           target) ||
          (target->has_condition() ? lower_instanceof(
```

Comment:
I'm afraid this would not necessarily be correct: it is intentional that the right-hand side of this "or" expression is executed even when the left-hand side evaluates to true. Now it might still work for `code` and `condition` likely aren't used together.

Now that is not to say that this `bitor` shouldn't be replaced: I think it just requires a slightly larger change to make sure both operands are evaluated.